Swagger docs are not being generated

Due to 'bug' in go-swagger, swagger docs are currently not being generated.

The main reason is usage /golang.org/x/crypto package, which has Linter/Code issues. Go-swagger complains with error:

```
swagger generate spec -o ./swagger.json --scan-models
/usr/local/go/src/crypto/x509/root_cgo_darwin.go:210:30: cannot convert nil (untyped nil value) to _Ctype_CFDataRef
/usr/local/go/src/crypto/x509/root_cgo_darwin.go:211:39: cannot convert nil (untyped nil value) to _Ctype_CFDataRef
/usr/local/go/src/crypto/x509/root_cgo_darwin.go:221:22: cannot convert nil (untyped nil value) to _Ctype_CFDataRef
/Users/ribice/go/src/github.com/ugorji/go/codec/goversion_makemap_gte_go19.go:12:10: MakeMapWithSize not declared by package reflect
/Users/ribice/go/src/github.com/ugorji/go/codec/goversion_makemap_gte_go19.go:14:9: MakeMapWithSize not declared by package reflect
couldn't load packages due to errors: github.com/ugorji/go/codec, crypto/x509
```

Per author's words:

I think it migh tmake it consider less packages
otherwise the error is probably a setting here: `https://github.com/go-swagger/go-swagger/blob/master/scan/scanner.go#L205-L216`
or a bug here: `https://github.com/go-swagger/go-swagger/blob/master/scan/classifier.go#L72:30`

Also I installed go-swagger using brew, which downloads latest release (6 months old). Perhaps download from master using go-get and running the binary would solve the issue.

--- a/cmd/api/server.go
+++ b/cmd/api/server.go
@@ -35,7 +35,6 @@ package main
 import (
 	"io/ioutil"
 	"log"
-	"net/http"
 
 	"github.com/gin-contrib/cors"
 
@@ -46,6 +45,7 @@ import (
 	"github.com/ribice/gorsk/cmd/api/config"
 	"github.com/ribice/gorsk/cmd/api/mw"
 	"github.com/ribice/gorsk/cmd/api/service"
+	_ "github.com/ribice/gorsk/cmd/api/swagger"
 	"github.com/ribice/gorsk/internal/account"
 	"github.com/ribice/gorsk/internal/auth"
 	"github.com/ribice/gorsk/internal/rbac"
@@ -92,7 +92,7 @@ func addV1Services(cfg *config.Configuration, r *gin.Engine, db *pg.DB, log *zap
 func docHandler(c *gin.Context) {
 	c.Header("Content-Type", "application/json")
 	data, _ := ioutil.ReadFile("swagger.json")
-	c.JSON(http.StatusOK, data)
+	c.Writer.Write(data)
 }
 
 func checkErr(err error) {

--- a/cmd/api/service/account.go
+++ b/cmd/api/service/account.go
@@ -23,7 +23,7 @@ func NewAccount(svc *account.Service, r *gin.RouterGroup) {
 	a := Account{svc: svc}
 	ar := r.Group("/users")
 	// swagger:route POST /users users accCreate
-	// Creates a new user account.s
+	// Creates new user account.
 	// responses:
 	//  200: userResp
 	//  400: errMsg

--- a/cmd/api/service/user.go
+++ b/cmd/api/service/user.go
@@ -24,8 +24,7 @@ func NewUser(svc *user.Service, r *gin.RouterGroup) {
 	// swagger:operation GET /users users listUsers
 	// ---
 	// summary: Returns list of users.
-	// description: Returns list of users. Depending on the user role requesting it, it may return all users for
-	// SuperAdmin/Admin users, all company/location users for Company/Location admins, and an error for non-admin users.
+	// description: Returns list of users. Depending on the user role requesting it, it may return all users for SuperAdmin/Admin users, all company/location users for Company/Location admins, and an error for non-admin users.
 	// parameters:
 	// - name: limit
 	//   in: query
@@ -76,7 +75,7 @@ func NewUser(svc *user.Service, r *gin.RouterGroup) {
 	// swagger:operation PATCH /users/{id} users userUpdate
 	// ---
 	// summary: Updates user's contact information
-	// description: Updates user's contact information: first name, last name, mobile, phone, address
+	// description: Updates user's contact information -> first name, last name, mobile, phone, address.
 	// parameters:
 	// - name: id
 	//   in: path

--- a/cmd/api/swagger.json
+++ b/cmd/api/swagger.json
@@ -1,0 +1,643 @@
+{
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "schemes": [
+    "http"
+  ],
+  "swagger": "2.0",
+  "info": {
+    "description": "API Docs for GORSK v1",
+    "title": "GORSK - Go(lang) restful starter kit",
+    "contact": {
+      "name": "Emir Ribic",
+      "url": "https://ribice.ba",
+      "email": "ribice@gmail.com"
+    },
+    "license": {
+      "name": "MIT",
+      "url": "http://opensource.org/licenses/MIT"
+    },
+    "version": "1.0.0"
+  },
+  "host": "localhost:8001",
+  "basePath": "/v1",
+  "paths": {
+    "/login": {
+      "post": {
+        "tags": [
+          "auth"
+        ],
+        "summary": "Logs in user by username and password.",
+        "operationId": "login",
+        "parameters": [
+          {
+            "name": "Body",
+            "in": "body",
+            "schema": {
+              "$ref": "#/definitions/Credentials"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "$ref": "#/responses/loginResp"
+          },
+          "400": {
+            "$ref": "#/responses/errMsg"
+          },
+          "401": {
+            "$ref": "#/responses/errMsg"
+          },
+          "404": {
+            "$ref": "#/responses/errMsg"
+          },
+          "500": {
+            "$ref": "#/responses/err"
+          }
+        }
+      }
+    },
+    "/users": {
+      "get": {
+        "description": "Returns list of users. Depending on the user role requesting it, it may return all users for SuperAdmin/Admin users, all company/location users for Company/Location admins, and an error for non-admin users.",
+        "tags": [
+          "users"
+        ],
+        "summary": "Returns list of users.",
+        "operationId": "listUsers",
+        "parameters": [
+          {
+            "type": "int",
+            "description": "number of results",
+            "name": "limit",
+            "in": "query"
+          },
+          {
+            "type": "int",
+            "description": "page number",
+            "name": "page",
+            "in": "query"
+          }
+        ],
+        "responses": {
+          "200": {
+            "$ref": "#/responses/userListResp"
+          },
+          "400": {
+            "$ref": "#/responses/errMsg"
+          },
+          "401": {
+            "$ref": "#/responses/err"
+          },
+          "403": {
+            "$ref": "#/responses/err"
+          },
+          "500": {
+            "$ref": "#/responses/errS"
+          }
+        }
+      },
+      "post": {
+        "tags": [
+          "users"
+        ],
+        "summary": "Creates new user account.",
+        "operationId": "accCreate",
+        "parameters": [
+          {
+            "name": "Body",
+            "in": "body",
+            "schema": {
+              "$ref": "#/definitions/Register"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "$ref": "#/responses/userResp"
+          },
+          "400": {
+            "$ref": "#/responses/errMsg"
+          },
+          "401": {
+            "$ref": "#/responses/err"
+          },
+          "403": {
+            "$ref": "#/responses/errMsg"
+          },
+          "500": {
+            "$ref": "#/responses/err"
+          }
+        }
+      }
+    },
+    "/users/{id}": {
+      "get": {
+        "description": "Returns a single user by its ID.",
+        "tags": [
+          "users"
+        ],
+        "summary": "Returns a single user.",
+        "operationId": "getUser",
+        "parameters": [
+          {
+            "type": "int",
+            "description": "id of user",
+            "name": "id",
+            "in": "path",
+            "required": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "$ref": "#/responses/userResp"
+          },
+          "400": {
+            "$ref": "#/responses/err"
+          },
+          "401": {
+            "$ref": "#/responses/err"
+          },
+          "403": {
+            "$ref": "#/responses/err"
+          },
+          "404": {
+            "$ref": "#/responses/err"
+          },
+          "500": {
+            "$ref": "#/responses/err"
+          }
+        }
+      },
+      "delete": {
+        "description": "Deletes a user with requested ID.",
+        "tags": [
+          "users"
+        ],
+        "summary": "Deletes a user",
+        "operationId": "userDelete",
+        "parameters": [
+          {
+            "type": "int",
+            "description": "id of user",
+            "name": "id",
+            "in": "path",
+            "required": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "$ref": "#/responses/ok"
+          },
+          "400": {
+            "$ref": "#/responses/err"
+          },
+          "401": {
+            "$ref": "#/responses/err"
+          },
+          "403": {
+            "$ref": "#/responses/err"
+          },
+          "500": {
+            "$ref": "#/responses/err"
+          }
+        }
+      },
+      "patch": {
+        "description": "Updates user's contact information -\u003e first name, last name, mobile, phone, address.",
+        "tags": [
+          "users"
+        ],
+        "summary": "Updates user's contact information",
+        "operationId": "userUpdate",
+        "parameters": [
+          {
+            "type": "int",
+            "description": "id of user",
+            "name": "id",
+            "in": "path",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/UpdateUser"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "$ref": "#/responses/userResp"
+          },
+          "400": {
+            "$ref": "#/responses/errMsg"
+          },
+          "401": {
+            "$ref": "#/responses/err"
+          },
+          "403": {
+            "$ref": "#/responses/err"
+          },
+          "500": {
+            "$ref": "#/responses/err"
+          }
+        }
+      }
+    },
+    "/users/{id}/password": {
+      "patch": {
+        "description": "If user's old passowrd is correct, it will be replaced with new password.",
+        "tags": [
+          "users"
+        ],
+        "summary": "Changes user's password.",
+        "operationId": "pwChange",
+        "parameters": [
+          {
+            "type": "int",
+            "description": "id of user",
+            "name": "id",
+            "in": "path",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Password"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "$ref": "#/responses/ok"
+          },
+          "400": {
+            "$ref": "#/responses/errMsg"
+          },
+          "401": {
+            "$ref": "#/responses/err"
+          },
+          "403": {
+            "$ref": "#/responses/err"
+          },
+          "500": {
+            "$ref": "#/responses/err"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AccessRole": {
+      "description": "AccessRole represents access role type",
+      "type": "integer",
+      "format": "int8",
+      "x-go-package": "github.com/ribice/gorsk/internal"
+    },
+    "Credentials": {
+      "description": "Credentials contains login request",
+      "type": "object",
+      "properties": {
+        "password": {
+          "type": "string",
+          "x-go-name": "Password"
+        },
+        "username": {
+          "type": "string",
+          "x-go-name": "Username"
+        }
+      },
+      "x-go-package": "github.com/ribice/gorsk/cmd/api/request"
+    },
+    "Password": {
+      "description": "Password contains password change request",
+      "type": "object",
+      "properties": {
+        "new_password": {
+          "type": "string",
+          "x-go-name": "NewPassword"
+        },
+        "new_password_confirm": {
+          "type": "string",
+          "x-go-name": "NewPasswordConfirm"
+        },
+        "old_password": {
+          "type": "string",
+          "x-go-name": "OldPassword"
+        }
+      },
+      "x-go-package": "github.com/ribice/gorsk/cmd/api/request"
+    },
+    "Register": {
+      "description": "Register contains registration request",
+      "type": "object",
+      "properties": {
+        "company_id": {
+          "type": "integer",
+          "format": "int64",
+          "x-go-name": "CompanyID"
+        },
+        "email": {
+          "type": "string",
+          "x-go-name": "Email"
+        },
+        "first_name": {
+          "type": "string",
+          "x-go-name": "FirstName"
+        },
+        "last_name": {
+          "type": "string",
+          "x-go-name": "LastName"
+        },
+        "location_id": {
+          "type": "integer",
+          "format": "int64",
+          "x-go-name": "LocationID"
+        },
+        "password": {
+          "type": "string",
+          "x-go-name": "Password"
+        },
+        "password_confirm": {
+          "type": "string",
+          "x-go-name": "PasswordConfirm"
+        },
+        "role_id": {
+          "type": "integer",
+          "format": "int64",
+          "x-go-name": "RoleID"
+        },
+        "username": {
+          "type": "string",
+          "x-go-name": "Username"
+        }
+      },
+      "x-go-package": "github.com/ribice/gorsk/cmd/api/request"
+    },
+    "Role": {
+      "description": "Role model",
+      "type": "object",
+      "properties": {
+        "access_level": {
+          "$ref": "#/definitions/AccessRole"
+        },
+        "id": {
+          "type": "integer",
+          "format": "int64",
+          "x-go-name": "ID"
+        },
+        "name": {
+          "type": "string",
+          "x-go-name": "Name"
+        }
+      },
+      "x-go-package": "github.com/ribice/gorsk/internal"
+    },
+    "UpdateUser": {
+      "description": "UpdateUser contains user update data from json request",
+      "type": "object",
+      "properties": {
+        "address": {
+          "type": "string",
+          "x-go-name": "Address"
+        },
+        "first_name": {
+          "type": "string",
+          "x-go-name": "FirstName"
+        },
+        "last_name": {
+          "type": "string",
+          "x-go-name": "LastName"
+        },
+        "mobile": {
+          "type": "string",
+          "x-go-name": "Mobile"
+        },
+        "phone": {
+          "type": "string",
+          "x-go-name": "Phone"
+        }
+      },
+      "x-go-package": "github.com/ribice/gorsk/cmd/api/request"
+    },
+    "User": {
+      "description": "User represents user domain model",
+      "type": "object",
+      "properties": {
+        "active": {
+          "type": "boolean",
+          "x-go-name": "Active"
+        },
+        "address": {
+          "type": "string",
+          "x-go-name": "Address"
+        },
+        "company_id": {
+          "type": "integer",
+          "format": "int64",
+          "x-go-name": "CompanyID"
+        },
+        "created_at": {
+          "type": "string",
+          "format": "date-time",
+          "x-go-name": "CreatedAt"
+        },
+        "deleted_at": {
+          "type": "string",
+          "format": "date-time",
+          "x-go-name": "DeletedAt"
+        },
+        "email": {
+          "type": "string",
+          "x-go-name": "Email"
+        },
+        "first_name": {
+          "type": "string",
+          "x-go-name": "FirstName"
+        },
+        "id": {
+          "type": "integer",
+          "format": "int64",
+          "x-go-name": "ID"
+        },
+        "last_login": {
+          "type": "string",
+          "format": "date-time",
+          "x-go-name": "LastLogin"
+        },
+        "last_name": {
+          "type": "string",
+          "x-go-name": "LastName"
+        },
+        "location_id": {
+          "type": "integer",
+          "format": "int64",
+          "x-go-name": "LocationID"
+        },
+        "mobile": {
+          "type": "string",
+          "x-go-name": "Mobile"
+        },
+        "phone": {
+          "type": "string",
+          "x-go-name": "Phone"
+        },
+        "role": {
+          "$ref": "#/definitions/Role"
+        },
+        "updated_at": {
+          "type": "string",
+          "format": "date-time",
+          "x-go-name": "UpdatedAt"
+        },
+        "username": {
+          "type": "string",
+          "x-go-name": "Username"
+        }
+      },
+      "x-go-package": "github.com/ribice/gorsk/internal"
+    }
+  },
+  "responses": {
+    "err": {
+      "description": "Error response"
+    },
+    "errMsg": {
+      "description": "Error response with message",
+      "headers": {
+        "message": {
+          "type": "string"
+        }
+      }
+    },
+    "loginResp": {
+      "description": "Login response",
+      "schema": {
+        "type": "object",
+        "properties": {
+          "expires": {
+            "type": "string",
+            "x-go-name": "Expires"
+          },
+          "token": {
+            "type": "string",
+            "x-go-name": "Token"
+          }
+        }
+      }
+    },
+    "ok": {
+      "description": "Success response"
+    },
+    "userListResp": {
+      "description": "Users model response",
+      "schema": {
+        "type": "object",
+        "properties": {
+          "page": {
+            "type": "integer",
+            "format": "int64",
+            "x-go-name": "Page"
+          },
+          "users": {
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/User"
+            },
+            "x-go-name": "Users"
+          }
+        }
+      }
+    },
+    "userResp": {
+      "description": "User model response",
+      "schema": {
+        "type": "object",
+        "properties": {
+          "active": {
+            "type": "boolean",
+            "x-go-name": "Active"
+          },
+          "address": {
+            "type": "string",
+            "x-go-name": "Address"
+          },
+          "company_id": {
+            "type": "integer",
+            "format": "int64",
+            "x-go-name": "CompanyID"
+          },
+          "created_at": {
+            "type": "string",
+            "format": "date-time",
+            "x-go-name": "CreatedAt"
+          },
+          "deleted_at": {
+            "type": "string",
+            "format": "date-time",
+            "x-go-name": "DeletedAt"
+          },
+          "email": {
+            "type": "string",
+            "x-go-name": "Email"
+          },
+          "first_name": {
+            "type": "string",
+            "x-go-name": "FirstName"
+          },
+          "id": {
+            "type": "integer",
+            "format": "int64",
+            "x-go-name": "ID"
+          },
+          "last_login": {
+            "type": "string",
+            "format": "date-time",
+            "x-go-name": "LastLogin"
+          },
+          "last_name": {
+            "type": "string",
+            "x-go-name": "LastName"
+          },
+          "location_id": {
+            "type": "integer",
+            "format": "int64",
+            "x-go-name": "LocationID"
+          },
+          "mobile": {
+            "type": "string",
+            "x-go-name": "Mobile"
+          },
+          "phone": {
+            "type": "string",
+            "x-go-name": "Phone"
+          },
+          "role": {
+            "$ref": "#/definitions/Role"
+          },
+          "updated_at": {
+            "type": "string",
+            "format": "date-time",
+            "x-go-name": "UpdatedAt"
+          },
+          "username": {
+            "type": "string",
+            "x-go-name": "Username"
+          }
+        }
+      }
+    }
+  },
+  "securityDefinitions": {
+    "bearer": {
+      "type": "apiKey",
+      "name": "Authorization",
+      "in": "header"
+    }
+  },
+  "security": [
+    {
+      "bearer": [
+        "[]"
+      ]
+    }
+  ]
+}

